src/extension.ts

```diff
@@ -18,8 +18,12 @@ export function activate(context: vscode.ExtensionContext) {
     let sortOnCommandDisposer = vscode.commands.registerCommand('extension.sortTypescriptImports', () => {
         // The code you place here will be executed every time your command is executed
 
-        if (vscode.window.activeTextEditor.document.languageId === 'typescript'
-            || vscode.window.activeTextEditor.document.languageId === 'typescriptreact') {
+        if (shouldEnableJavascript() && (
```

Comment:
I think you need an extra set of parentheses here:

```ts
if ((shouldEnableJavascript() && (
                 vscode.window.activeTextEditor.document.languageId === 'javascript' ||
                 vscode.window.activeTextEditor.document.languageId === 'javascriptreact'
             )) ||
```

Alternatively, it may be worthwhile to refactor these to functions.

Reply:
I don't think you need the extra parenthesis from a logical standpoint... is that just a style preference?
I also broke down this if statement into sub functions for readability.